<a href="https://colab.research.google.com/github/mateus-neres/tratamento_enem_2019/blob/main/tratamento_enem_2019.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**PROJETO DE ANÁLISE ESTATISTICA ENEM 2019**

Este projeto analisa os resultados do ENEM no estado de São Paulo no ano de 2019.

Os dados foram extraídos do site do INEP:

http://inep.gov.br/mocrodados

In [49]:
import numpy as np
import pandas as pd

In [50]:
dados = pd.read_csv("/content/drive/MyDrive/curso_Udemy/curso_estatistica_python/microdados_enem_2019_sp.csv", sep=";", encoding="iso-8859-1" )
# encoding: codificação de caracteres, normalmente ultiza-se o "iso-8859-1", utf-8, latin-1

#**Destrucando colunas**

Exibi todas as linhas e colunas sem ocutações ou abreviações

In [51]:
pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", None)

#**Estração e limpesa dos dados**


In [52]:
dados.head()

NU_INSCRICAO  CO_MUNICIPIO_RESIDENCIA NO_MUNICIPIO_RESIDENCIA  \
0     1,90E+11                  3546801            Santa Isabel   
1     1,90E+11                  3538204             Pinhalzinho   
2     1,90E+11                  3550308               São Paulo   
3     1,90E+11                  3548708   São Bernardo do Campo   
4     1,90E+11                  3549409    São Joaquim da Barra   

   CO_UF_RESIDENCIA SG_UF_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0                35               SP        22       M                1   
1                35               SP        19       M                1   
2                35               SP        17       F                1   
3                35               SP        19       M                1   
4                35               SP        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  CO_MUNICIPIO_NASCIMENTO  \
0            3                 1                1506500.0   
1            3                 1                1503804.0   
2            3                 1                1501402.0   
3            3                 1                3548708.0   
4            1                 1                3549409.0   

  NO_MUNICIPIO_NASCIMENTO  CO_UF_NASCIMENTO SG_UF_NASCIMENTO  TP_ST_CONCLUSAO  \
0    Santa Izabel do Pará              15.0               PA                1   
1                 Jacundá              15.0               PA                1   
2                   Belém              15.0               PA                2   
3   São Bernardo do Campo              35.0               SP                2   
4    São Joaquim da Barra              35.0               SP                1   

   TP_ANO_CONCLUIU  TP_ESCOLA  TP_ENSINO  IN_TREINEIRO   CO_ESCOLA  \
0                4          1        1.0             0         NaN   
1                2          1        1.0             0         NaN   
2                0          3        1.0             0  15567761.0   
3                0          2        1.0             0  35904958.0   
4                3          1        1.0             0         NaN   

   CO_MUNICIPIO_ESC       NO_MUNICIPIO_ESC  CO_UF_ESC SG_UF_ESC  \
0               NaN                    NaN        NaN       NaN   
1               NaN                    NaN        NaN       NaN   
2         1501402.0                  Belém       15.0        PA   
3         3548708.0  São Bernardo do Campo       35.0        SP   
4               NaN                    NaN        NaN       NaN   

   TP_DEPENDENCIA_ADM_ESC  TP_LOCALIZACAO_ESC  TP_SIT_FUNC_ESC  \
0                     NaN                 NaN              NaN   
1                     NaN                 NaN              NaN   
2                     4.0                 1.0              1.0   
3                     2.0                 1.0              1.0   
4                     NaN                 NaN              NaN   

   TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  \
0               1               1               1               1      5646.0   
1               0               0               0               0         NaN   
2               1               1               1               1      5762.0   
3               1               1               1               1      4022.0   
4               0               0               0               0         NaN   

   NU_NOTA_CH  NU_NOTA_LC  NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  \
0      5858.0      5929.0      6002.0          0                1.0   
1         NaN         NaN         NaN          1                NaN   
2      6411.0      6343.0      7318.0          0                1.0   
3      4248.0      5096.0      3932.0          1                1.0   
4         NaN         NaN         NaN          1                NaN   

   NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  \
0          160.0          200.0          180.0          200.0          200.0   

In [53]:
#O número total de inscrito em são Paulo foi de 813772, sendo de 15,97% de todo o Brasil que foi de 50955308 inscritos
dados.shape # vai passar (813772, 44) -> (linhas, registros ou atributos)

(813772, 44)

In [54]:
#Ánalise dos tipos de atributos
#Object: string
#int64: inteiros
#float64: reais
#complex: complexos
dados.dtypes


NU_INSCRICAO                object
CO_MUNICIPIO_RESIDENCIA      int64
NO_MUNICIPIO_RESIDENCIA     object
CO_UF_RESIDENCIA             int64
SG_UF_RESIDENCIA            object
NU_IDADE                     int64
TP_SEXO                     object
TP_ESTADO_CIVIL              int64
TP_COR_RACA                  int64
TP_NACIONALIDADE             int64
CO_MUNICIPIO_NASCIMENTO    float64
NO_MUNICIPIO_NASCIMENTO     object
CO_UF_NASCIMENTO           float64
SG_UF_NASCIMENTO            object
TP_ST_CONCLUSAO              int64
TP_ANO_CONCLUIU              int64
TP_ESCOLA                    int64
TP_ENSINO                  float64
IN_TREINEIRO                 int64
CO_ESCOLA                  float64
CO_MUNICIPIO_ESC           float64
NO_MUNICIPIO_ESC            object
CO_UF_ESC                  float64
SG_UF_ESC                   object
TP_DEPENDENCIA_ADM_ESC     float64
TP_LOCALIZACAO_ESC         float64
TP_SIT_FUNC_ESC            float64
TP_PRESENCA_CN               int64
TP_PRESENCA_CH      

Removendo colunas desnecessárias

In [55]:
dados1 = dados.drop(columns=["CO_UF_RESIDENCIA","SG_UF_RESIDENCIA","CO_MUNICIPIO_NASCIMENTO","CO_MUNICIPIO_RESIDENCIA","NO_MUNICIPIO_NASCIMENTO","CO_UF_NASCIMENTO","SG_UF_NASCIMENTO","TP_ANO_CONCLUIU", "TP_ENSINO","CO_MUNICIPIO_ESC","CO_UF_ESC", "SG_UF_ESC","TP_DEPENDENCIA_ADM_ESC","TP_LOCALIZACAO_ESC","TP_SIT_FUNC_ESC"], )

In [56]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

    CO_ESCOLA       NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  \
0         NaN                    NaN               1               1   
1         NaN                    NaN               0               0   
2  15567761.0                  Belém               1               1   
3  35904958.0  São Bernardo do Campo               1               1   
4         NaN                    NaN               0               0   

   TP_PRESENCA_LC  TP_PRESENCA_MT  NU_NOTA_CN  NU_NOTA_CH  NU_NOTA_LC  \
0               1               1      5646.0      5858.0      5929.0   
1               0               0         NaN         NaN         NaN   
2               1               1      5762.0      6411.0      6343.0   
3               1               1      4022.0      4248.0      5096.0   
4               0               0         NaN         NaN         NaN   

   NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0      6002.0          0                1.0          160.0          200.0   
1         NaN          1                NaN            NaN            NaN   
2      7318.0          0                1.0          160.0          120.0   
3      3932.0          1                1.0           80.0          160.0   
4         NaN          1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO Q025  
0          180.0          200.0          200.0            940.0    B  
1            NaN            NaN            NaN              NaN    A  
2          200.0          200.0          200.0            880.0    A  
3          120.0          100.0          100.0            560.0    B  
4            NaN            NaN            NaN              NaN    B

In [57]:
dados1.shape

(813772, 29)

#**Renomear varias colunas**

In [58]:
# Aplicando em apenas uma coluna
dados1 = dados1.rename(columns={"NU_NOTA_REDACAO":"NOTA_REDACAO"})

In [59]:
# Aplicando em varias colunas
dados1 = dados1.rename(columns={"NU_NOTA_CN":"NOTA_CN", "NU_NOTA_CH":"NOTA_CH",
                                "NU_NOTA_LC":"NOTA_LC", "NU_NOTA_MT":"NOTA_MT"})

In [60]:
dados1 = dados1.rename(columns={"NU_INSCRICAO":"INSCRICAO","NO_MUNICIPIO_RESIDENCIA":"MUNICIPIO_RESIDENCIA",
                                "NU_IDADE":"IDADE", "TP_SEXO":"SEXO","TP_ESTADO_CIVIL":"ESTADO_CIVIL",
                                "TP_COR_RACA":"RACA", "TP_NACIONALIDADE":"NACIONALIDADE","TP_ESCOLA":"ESCOLA","Q025":"INTERNET"})

In [61]:
dados1 = dados1.rename(columns={"NU_NOTA_COMP1":"COMP1","NU_NOTA_COMP2":"COMP2","NU_NOTA_COMP3":"COMP3","NU_NOTA_COMP4":"COMP4","NU_NOTA_COMP5	":"COMP5"})

In [62]:
dados1 = dados1.rename(columns={"TP_LINGUA":"LINGUA"})

In [63]:
dados1.head()


INSCRICAO   MUNICIPIO_RESIDENCIA  IDADE SEXO  ESTADO_CIVIL  RACA  \
0  1,90E+11           Santa Isabel     22    M             1     3   
1  1,90E+11            Pinhalzinho     19    M             1     3   
2  1,90E+11              São Paulo     17    F             1     3   
3  1,90E+11  São Bernardo do Campo     19    M             1     3   
4  1,90E+11   São Joaquim da Barra     37    M             2     1   

   NACIONALIDADE  TP_ST_CONCLUSAO  ESCOLA  IN_TREINEIRO   CO_ESCOLA  \
0              1                1       1             0         NaN   
1              1                1       1             0         NaN   
2              1                2       3             0  15567761.0   
3              1                2       2             0  35904958.0   
4              1                1       1             0         NaN   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
1                    NaN               0               0               0   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
4                    NaN               0               0               0   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT  LINGUA  \
0               1   5646.0   5858.0   5929.0   6002.0       0   
1               0      NaN      NaN      NaN      NaN       1   
2               1   5762.0   6411.0   6343.0   7318.0       0   
3               1   4022.0   4248.0   5096.0   3932.0       1   
4               0      NaN      NaN      NaN      NaN       1   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  NU_NOTA_COMP5  NOTA_REDACAO  \
0                1.0  160.0  200.0  180.0  200.0          200.0         940.0   
1                NaN    NaN    NaN    NaN    NaN            NaN           NaN   
2                1.0  160.0  120.0  200.0  200.0          200.0         880.0   
3                1.0   80.0  160.0  120.0  100.0          100.0         560.0   
4                NaN    NaN    NaN    NaN    NaN            NaN           NaN   

  INTERNET  
0        B  
1        A  
2        A  
3        B  
4        B

#**Conrrigindo erro nas notas individuais**

In [64]:
dados1.loc[:,"NOTA_CN"] /= 10
dados1.loc[:,"NOTA_CH"] /= 10
dados1.loc[:,"NOTA_LC"] /= 10
dados1.loc[:,"NOTA_MT"] /= 10

In [65]:
display(dados1.head(5))

INSCRICAO   MUNICIPIO_RESIDENCIA  IDADE SEXO  ESTADO_CIVIL  RACA  \
0  1,90E+11           Santa Isabel     22    M             1     3   
1  1,90E+11            Pinhalzinho     19    M             1     3   
2  1,90E+11              São Paulo     17    F             1     3   
3  1,90E+11  São Bernardo do Campo     19    M             1     3   
4  1,90E+11   São Joaquim da Barra     37    M             2     1   

   NACIONALIDADE  TP_ST_CONCLUSAO  ESCOLA  IN_TREINEIRO   CO_ESCOLA  \
0              1                1       1             0         NaN   
1              1                1       1             0         NaN   
2              1                2       3             0  15567761.0   
3              1                2       2             0  35904958.0   
4              1                1       1             0         NaN   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
1                    NaN               0               0               0   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
4                    NaN               0               0               0   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT  LINGUA  \
0               1    564.6    585.8    592.9    600.2       0   
1               0      NaN      NaN      NaN      NaN       1   
2               1    576.2    641.1    634.3    731.8       0   
3               1    402.2    424.8    509.6    393.2       1   
4               0      NaN      NaN      NaN      NaN       1   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  NU_NOTA_COMP5  NOTA_REDACAO  \
0                1.0  160.0  200.0  180.0  200.0          200.0         940.0   
1                NaN    NaN    NaN    NaN    NaN            NaN           NaN   
2                1.0  160.0  120.0  200.0  200.0          200.0         880.0   
3                1.0   80.0  160.0  120.0  100.0          100.0         560.0   
4                NaN    NaN    NaN    NaN    NaN            NaN           NaN   

  INTERNET  
0        B  
1        A  
2        A  
3        B  
4        B

##**Reanomear registros (linha)**

In [66]:
dados1["RACA"] = dados1["RACA"].replace({0:"não_declarado", 1:"branca", 2:"preta",3:"parda",4:"amarela",5:"indígena"})
dados1["INTERNET"] = dados1["INTERNET"].replace({"A":"não", "B":"sim"})
# Outra maneira de fazer:
# Exp: dados1["INTERNET"] = dados1["INTERNET"].replace(["A","B"], ["não","sim"])

In [67]:
dados1["ESCOLA"] = dados1["ESCOLA"].replace([1,2,3,4], ["não_respondeu","pública","privada","exterior"])

In [68]:
dados1["LINGUA"] = dados1["LINGUA"].replace([0,1],["inglês","espanhol"])

In [69]:
dados1.head()

INSCRICAO   MUNICIPIO_RESIDENCIA  IDADE SEXO  ESTADO_CIVIL    RACA  \
0  1,90E+11           Santa Isabel     22    M             1   parda   
1  1,90E+11            Pinhalzinho     19    M             1   parda   
2  1,90E+11              São Paulo     17    F             1   parda   
3  1,90E+11  São Bernardo do Campo     19    M             1   parda   
4  1,90E+11   São Joaquim da Barra     37    M             2  branca   

   NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO   CO_ESCOLA  \
0              1                1  não_respondeu             0         NaN   
1              1                1  não_respondeu             0         NaN   
2              1                2        privada             0  15567761.0   
3              1                2        pública             0  35904958.0   
4              1                1  não_respondeu             0         NaN   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
1                    NaN               0               0               0   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
4                    NaN               0               0               0   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT    LINGUA  \
0               1    564.6    585.8    592.9    600.2    inglês   
1               0      NaN      NaN      NaN      NaN  espanhol   
2               1    576.2    641.1    634.3    731.8    inglês   
3               1    402.2    424.8    509.6    393.2  espanhol   
4               0      NaN      NaN      NaN      NaN  espanhol   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  NU_NOTA_COMP5  NOTA_REDACAO  \
0                1.0  160.0  200.0  180.0  200.0          200.0         940.0   
1                NaN    NaN    NaN    NaN    NaN            NaN           NaN   
2                1.0  160.0  120.0  200.0  200.0          200.0         880.0   
3                1.0   80.0  160.0  120.0  100.0          100.0         560.0   
4                NaN    NaN    NaN    NaN    NaN            NaN           NaN   

  INTERNET  
0      sim  
1      não  
2      não  
3      sim  
4      sim

#**Análise exploratória e Transformação dos dados**

**Análise das idades**

In [70]:
dados1["IDADE"].value_counts()

17    219313
18    147548
19     81236
16     59316
20     53009
21     37242
22     29489
23     23531
24     18851
25     15103
26     12465
15     11583
27     10367
28      8986
29      7650
30      6967
31      6331
32      5685
33      5175
34      4795
35      4340
36      4109
37      4105
38      3767
39      3299
40      3058
41      2638
42      2462
43      2307
44      2069
45      1939
46      1718
47      1578
48      1496
49      1304
50      1167
51      1014
52       933
53       841
54       727
55       652
56       530
57       497
58       388
14       355
59       325
60       254
61       236
62       176
63       144
64       139
65       115
66        81
67        72
68        53
69        43
70        40
13        29
71        25
72        17
73        16
74        13
75        10
76         8
12         6
8          6
79         4
77         3
81         3
7          3
78         3
5          2
6          2
82         2
80         2
86         1
4          1

In [71]:
dados1["IDADE"].value_counts().sort_index()

2          1
3          1
4          1
5          2
6          2
7          3
8          6
12         6
13        29
14       355
15     11583
16     59316
17    219313
18    147548
19     81236
20     53009
21     37242
22     29489
23     23531
24     18851
25     15103
26     12465
27     10367
28      8986
29      7650
30      6967
31      6331
32      5685
33      5175
34      4795
35      4340
36      4109
37      4105
38      3767
39      3299
40      3058
41      2638
42      2462
43      2307
44      2069
45      1939
46      1718
47      1578
48      1496
49      1304
50      1167
51      1014
52       933
53       841
54       727
55       652
56       530
57       497
58       388
59       325
60       254
61       236
62       176
63       144
64       139
65       115
66        81
67        72
68        53
69        43
70        40
71        25
72        17
73        16
74        13
75        10
76         8
77         3
78         3
79         4
80         2
81         3

**Alguns valores estranhos, idade menores que 12 e idade maiores que 80.**

In [72]:
# Inscrições abaixo de 12 anos, serão excluídas devido a alta probabilidade de erro.
menores_12 = dados1.query("IDADE < 12")["MUNICIPIO_RESIDENCIA"].value_counts()

In [73]:
menores_12

São Paulo                 5
Tambaú                    1
Ribeirão Preto            1
Bálsamo                   1
São Bernardo do Campo     1
Embu das Artes            1
Queiroz                   1
Santana de Parnaíba       1
Vargem Grande Paulista    1
Auriflama                 1
Araçatuba                 1
Campinas                  1
Name: MUNICIPIO_RESIDENCIA, dtype: int64

In [74]:
dados_maior_11 = dados1.loc[dados1.IDADE > 11]

In [75]:
dados_maior_11.head()

INSCRICAO   MUNICIPIO_RESIDENCIA  IDADE SEXO  ESTADO_CIVIL    RACA  \
0  1,90E+11           Santa Isabel     22    M             1   parda   
1  1,90E+11            Pinhalzinho     19    M             1   parda   
2  1,90E+11              São Paulo     17    F             1   parda   
3  1,90E+11  São Bernardo do Campo     19    M             1   parda   
4  1,90E+11   São Joaquim da Barra     37    M             2  branca   

   NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO   CO_ESCOLA  \
0              1                1  não_respondeu             0         NaN   
1              1                1  não_respondeu             0         NaN   
2              1                2        privada             0  15567761.0   
3              1                2        pública             0  35904958.0   
4              1                1  não_respondeu             0         NaN   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
1                    NaN               0               0               0   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
4                    NaN               0               0               0   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT    LINGUA  \
0               1    564.6    585.8    592.9    600.2    inglês   
1               0      NaN      NaN      NaN      NaN  espanhol   
2               1    576.2    641.1    634.3    731.8    inglês   
3               1    402.2    424.8    509.6    393.2  espanhol   
4               0      NaN      NaN      NaN      NaN  espanhol   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  NU_NOTA_COMP5  NOTA_REDACAO  \
0                1.0  160.0  200.0  180.0  200.0          200.0         940.0   
1                NaN    NaN    NaN    NaN    NaN            NaN           NaN   
2                1.0  160.0  120.0  200.0  200.0          200.0         880.0   
3                1.0   80.0  160.0  120.0  100.0          100.0         560.0   
4                NaN    NaN    NaN    NaN    NaN            NaN           NaN   

  INTERNET  
0      sim  
1      não  
2      não  
3      sim  
4      sim

In [76]:
dados_maior_11["IDADE"].value_counts().sort_index()

12         6
13        29
14       355
15     11583
16     59316
17    219313
18    147548
19     81236
20     53009
21     37242
22     29489
23     23531
24     18851
25     15103
26     12465
27     10367
28      8986
29      7650
30      6967
31      6331
32      5685
33      5175
34      4795
35      4340
36      4109
37      4105
38      3767
39      3299
40      3058
41      2638
42      2462
43      2307
44      2069
45      1939
46      1718
47      1578
48      1496
49      1304
50      1167
51      1014
52       933
53       841
54       727
55       652
56       530
57       497
58       388
59       325
60       254
61       236
62       176
63       144
64       139
65       115
66        81
67        72
68        53
69        43
70        40
71        25
72        17
73        16
74        13
75        10
76         8
77         3
78         3
79         4
80         2
81         3
82         2
86         1
91         1
Name: IDADE, dtype: int64

**Treineiro**

In [77]:
dados_maior_11.query("IN_TREINEIRO == 1")["IN_TREINEIRO"].value_counts()

1    93988
Name: IN_TREINEIRO, dtype: int64

In [78]:
treineiros = dados_maior_11.loc[dados_maior_11.IN_TREINEIRO == 1]

In [79]:
treineiros.head()

INSCRICAO MUNICIPIO_RESIDENCIA  IDADE SEXO  ESTADO_CIVIL    RACA  \
20   1,90E+11               Santos     15    F             1   parda   
21   1,90E+11             Panorama     16    F             1   parda   
106  1,90E+11            São Paulo     16    F             1  branca   
109  1,90E+11     Águas de Lindóia     16    M             1  branca   
159  1,90E+11     Águas de Lindóia     15    M             1  branca   

     NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  CO_ESCOLA  \
20               1                3  não_respondeu             1        NaN   
21               1                3  não_respondeu             1        NaN   
106              1                3  não_respondeu             1        NaN   
109              1                3  não_respondeu             1        NaN   
159              1                3  não_respondeu             1        NaN   

    NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
20               NaN               1               1               1   
21               NaN               0               0               0   
106              NaN               1               1               1   
109              NaN               0               0               0   
159              NaN               0               0               0   

     TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT    LINGUA  \
20                1    455.4    506.7    480.4    573.2    inglês   
21                0      NaN      NaN      NaN      NaN  espanhol   
106               1    424.5    517.7    489.6    532.6    inglês   
109               0      NaN      NaN      NaN      NaN  espanhol   
159               0      NaN      NaN      NaN      NaN    inglês   

     TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  NU_NOTA_COMP5  \
20                 1.0  140.0  120.0  120.0  100.0           80.0   
21                 NaN    NaN    NaN    NaN    NaN            NaN   
106                1.0  120.0  120.0  120.0  100.0           80.0   
109                NaN    NaN    NaN    NaN    NaN            NaN   
159                NaN    NaN    NaN    NaN    NaN            NaN   

     NOTA_REDACAO INTERNET  
20          560.0      sim  
21            NaN      não  
106         540.0      sim  
109           NaN      sim  
159           NaN      sim

In [80]:
treineiros.to_csv("treineiros_enem_2019_sp.csv", encoding = "iso-8859-1", index = False)

In [81]:
# Separação entre vestibulando e treineiros
vestibulandos = dados_maior_11.loc[dados_maior_11.IN_TREINEIRO == 0]

In [82]:
vestibulandos.head()

INSCRICAO   MUNICIPIO_RESIDENCIA  IDADE SEXO  ESTADO_CIVIL    RACA  \
0  1,90E+11           Santa Isabel     22    M             1   parda   
1  1,90E+11            Pinhalzinho     19    M             1   parda   
2  1,90E+11              São Paulo     17    F             1   parda   
3  1,90E+11  São Bernardo do Campo     19    M             1   parda   
4  1,90E+11   São Joaquim da Barra     37    M             2  branca   

   NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO   CO_ESCOLA  \
0              1                1  não_respondeu             0         NaN   
1              1                1  não_respondeu             0         NaN   
2              1                2        privada             0  15567761.0   
3              1                2        pública             0  35904958.0   
4              1                1  não_respondeu             0         NaN   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
1                    NaN               0               0               0   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
4                    NaN               0               0               0   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT    LINGUA  \
0               1    564.6    585.8    592.9    600.2    inglês   
1               0      NaN      NaN      NaN      NaN  espanhol   
2               1    576.2    641.1    634.3    731.8    inglês   
3               1    402.2    424.8    509.6    393.2  espanhol   
4               0      NaN      NaN      NaN      NaN  espanhol   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  NU_NOTA_COMP5  NOTA_REDACAO  \
0                1.0  160.0  200.0  180.0  200.0          200.0         940.0   
1                NaN    NaN    NaN    NaN    NaN            NaN           NaN   
2                1.0  160.0  120.0  200.0  200.0          200.0         880.0   
3                1.0   80.0  160.0  120.0  100.0          100.0         560.0   
4                NaN    NaN    NaN    NaN    NaN            NaN           NaN   

  INTERNET  
0      sim  
1      não  
2      não  
3      sim  
4      sim

In [83]:
vestibulandos.shape

(719768, 29)

**Análise das presenças**

In [84]:
vestibulandos["TP_PRESENCA_CN"].value_counts()

1    500805
0    218693
2       270
Name: TP_PRESENCA_CN, dtype: int64

In [85]:
vestibulandos["TP_PRESENCA_CH"].value_counts()

1    533142
0    185969
2       657
Name: TP_PRESENCA_CH, dtype: int64

In [86]:
vestibulandos["TP_PRESENCA_LC"].value_counts()

1    533142
0    185969
2       657
Name: TP_PRESENCA_LC, dtype: int64

In [87]:
vestibulandos["TP_PRESENCA_MT"].value_counts()

1    500805
0    218693
2       270
Name: TP_PRESENCA_MT, dtype: int64

###RESULMO PRESENÇA NAS PROVAS

CIÊNCIAS DA NATUREZA:
500805 Presentes,
218693 Faltaram e
270 Eliminados

CIÊNCIAS HUMANAS:
533142 Presentes,
185969 Faltaram e
657 Eliminados

LINGUAGEM E CÓDIGOS:
533142 Presentes,
185969 Faltaram e
657 Eliminados

MATEMÁTICAT:
500805 Presentes,
218693 Faltaram e
270 Eliminados


In [88]:
vestibulandos["TP_STATUS_REDACAO"].value_counts().sort_index()

1.0    519763
2.0       913
3.0      1417
4.0      5513
6.0      3626
7.0       543
8.0       828
9.0       539
Name: TP_STATUS_REDACAO, dtype: int64

**valores ausentes**

In [89]:
vestibulandos["IDADE"].isnull().sum()

0

In [90]:
provas = ["NOTA_REDACAO","NOTA_CN","NOTA_MT","NOTA_LC","NOTA_CH"]

In [91]:
# Registros NAN nas notas
vestibulandos [provas].isnull().sum()

NOTA_REDACAO    186626
NOTA_CN         218963
NOTA_MT         218963
NOTA_LC         186626
NOTA_CH         186626
dtype: int64

In [92]:
vestibulandos_classificados = vestibulandos.loc[vestibulandos.TP_PRESENCA_CH == 1]
vestibulandos_classificados = vestibulandos_classificados.loc[vestibulandos_classificados.TP_PRESENCA_MT == 1]

In [93]:
vestibulandos_classificados["TP_PRESENCA_MT"].value_counts()

1    499940
Name: TP_PRESENCA_MT, dtype: int64

In [94]:
vestibulandos_classificados["TP_PRESENCA_CH"].value_counts()

1    499940
Name: TP_PRESENCA_CH, dtype: int64

In [95]:
vestibulandos_classificados["TP_PRESENCA_LC"].value_counts()

1    499940
Name: TP_PRESENCA_LC, dtype: int64

In [96]:
vestibulandos_classificados["TP_PRESENCA_CN"].value_counts()

1    499940
Name: TP_PRESENCA_CN, dtype: int64